### ISSUE TITLE
crashes when directed to getflow.com

### ISSUE THREAD

**Issue by r-a-v-a-s** [QupZilla/qupzilla#1895]
The program will crash before the page loads.
https://www.getflow.com/

Note that the website uses http://electron.atom.io/ (or maybe it's their app)

I built from source and tried in debug mode,
but it crashes without fully terminating the app.

I'm guessing this might be a Qt issue.

OS X 10.11.4 / Qt 5.6

**Comment by cranes-bill** [comment]
No problem on Linux....

**Comment by Martii** [comment]
Crashes on Linux here with distro package of QupZilla:

``` sh-session
Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/538.1 (KHTML, like Gecko) QupZilla/1.8.9 Safari/538.1
```

**Comment by Martii** [comment]
So this issue has some useful info at least for a pre-built package of the webkit version:

``` sh-session
$ qupzilla
QupZilla: 1 extensions loaded
QPainter::restore: Unbalanced save/restore
QPainter::restore: Unbalanced save/restore
QPainter::restore: Unbalanced save/restore
QPainter::restore: Unbalanced save/restore
QPainter::restore: Unbalanced save/restore
QPainter::restore: Unbalanced save/restore
QPainter::restore: Unbalanced save/restore
QupZilla: Crashed :( Saving backtrace in /home/user/.config/qupzilla/crashlog ...
Backtrace successfully saved in /home/user/.config/qupzilla/crashlog/Crash-2016-03-27T08:25:42.txt
Aborted (core dumped)
$
```

```
Time: Sun Mar 27 08:25:42 2016
Qt version: 5.5.1 (compiled with 5.5.1)
QupZilla version: 1.8.9
WebKit version: 538.1

============== BACKTRACE ==============
#0: qupzilla() [0x4033ac]
#1: /lib64/libc.so.6(+0x34b80) [0x7fbb90830b80]
#2: /usr/lib64/libQt5Gui.so.5(_ZNK7QPixmap6isNullEv+0) [0x7fbb8f44f790]
#3: /usr/lib64/libQt5WebKit.so.5(+0x9ace1f) [0x7fbb92101e1f]
#4: /usr/lib64/libQt5WebKit.so.5(+0x88b529) [0x7fbb91fe0529]
#5: /usr/lib64/libQt5WebKit.so.5(+0x9a85d8) [0x7fbb920fd5d8]
#6: /usr/lib64/libQt5WebKit.so.5(+0x88ca4e) [0x7fbb91fe1a4e]
#7: /usr/lib64/libQt5WebKit.so.5(_ZN16QWebFrameAdapter20renderRelativeCoordsEP8QPainteriRK7QRegion+0x98) [0x7fbb91bea018]
#8: /usr/lib64/libQt5WebKitWidgets.so.5(_ZN9QWebFrame6renderEP8QPainter6QFlagsINS_11RenderLayerEERK7QRegion+0x9e) [0x7fbb8fe88c3e]
#9: /usr/lib64/libQt5WebKitWidgets.so.5(_ZN9QWebFrame6renderEP8QPainterRK7QRegion+0x11) [0x7fbb8fe88c61]
#10: /usr/lib64/libQt5WebKitWidgets.so.5(_ZN8QWebView10paintEventEP11QPaintEvent+0x58) [0x7fbb8fe93758]
#11: /usr/lib64/libQt5Widgets.so.5(_ZN7QWidget5eventEP6QEvent+0x2c0) [0x7fbb93e75e70]
#12: /usr/lib64/libQt5WebKitWidgets.so.5(_ZN8QWebView5eventEP6QEvent+0x74) [0x7fbb8fe94894]
#13: /usr/lib64/libQt5Widgets.so.5(_ZN19QApplicationPrivate13notify_helperEP7QObjectP6QEvent+0x8c) [0x7fbb93e351ec]
#14: /usr/lib64/libQt5Widgets.so.5(_ZN12QApplication6notifyEP7QObjectP6QEvent+0x390) [0x7fbb93e3a320]
#15: /usr/lib64/libQt5Core.so.5(_ZN16QCoreApplication14notifyInternalEP7QObjectP6QEvent+0x6d) [0x7fbb9131a99d]
#16: /usr/lib64/libQt5Widgets.so.5(_ZN14QWidgetPrivate14sendPaintEventERK7QRegion+0x39) [0x7fbb93e6f129]
#17: /usr/lib64/libQt5Widgets.so.5(_ZN14QWidgetPrivate10drawWidgetEP12QPaintDeviceRK7QRegionRK6QPointiP8QPainterP19QWidgetBackingStore+0x5de) [0x7fbb93e6f74e]
#18: /usr/lib64/libQt5Widgets.so.5(+0x162013) [0x7fbb93e42013]
#19: /usr/lib64/libQt5Widgets.so.5(+0x162290) [0x7fbb93e42290]
#20: /usr/lib64/libQt5Widgets.so.5(_ZN14QWidgetPrivate16syncBackingStoreEv+0x3f) [0x7fbb93e6186f]
#21: /usr/lib64/libQt5Widgets.so.5(_ZN7QWidget5eventEP6QEvent+0x390) [0x7fbb93e75f40]
#22: /usr/lib64/libQt5Widgets.so.5(_ZN11QMainWindow5eventEP6QEvent+0x15b) [0x7fbb93f86ddb]
#23: /usr/lib64/libQt5Widgets.so.5(_ZN19QApplicationPrivate13notify_helperEP7QObjectP6QEvent+0x8c) [0x7fbb93e351ec]
#24: /usr/lib64/libQt5Widgets.so.5(_ZN12QApplication6notifyEP7QObjectP6QEvent+0x390) [0x7fbb93e3a320]
#25: /usr/lib64/libQt5Core.so.5(_ZN16QCoreApplication14notifyInternalEP7QObjectP6QEvent+0x6d) [0x7fbb9131a99d]
#26: /usr/lib64/libQt5Core.so.5(_ZN23QCoreApplicationPrivate16sendPostedEventsEP7QObjectiP11QThreadData+0x225) [0x7fbb9131d625]
#27: /usr/lib64/libQt5Core.so.5(+0x2834e3) [0x7fbb913654e3]
#28: /lib64/libglib-2.0.so.0(g_main_context_dispatch+0x287) [0x7fbb8df330d7]
#29: /lib64/libglib-2.0.so.0(+0x49338) [0x7fbb8df33338]
#30: /lib64/libglib-2.0.so.0(g_main_context_iteration+0x2c) [0x7fbb8df333dc]
#31: /usr/lib64/libQt5Core.so.5(_ZN20QEventDispatcherGlib13processEventsE6QFlagsIN10QEventLoop17ProcessEventsFlagEE+0x57) [0x7fbb91365557]
#32: /usr/lib64/libQt5Core.so.5(_ZN10QEventLoop4execE6QFlagsINS_17ProcessEventsFlagEE+0xda) [0x7fbb913196ba]
#33: /usr/lib64/libQt5Core.so.5(_ZN16QCoreApplication4execEv+0x8d) [0x7fbb9132058d]
#34: qupzilla() [0x4029ff]
#35: /lib64/libc.so.6(__libc_start_main+0xf0) [0x7fbb9081bfc0]
#36: qupzilla() [0x402d1e]
```

**Comment by cranes-bill** [comment]
Works to me with WebKit as well, but with Qt4...

**Comment by r-a-v-a-s** [comment]
Both the downloaded version and the version I built from the master branch seem to be using the QtWebKit:

```
Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x0000000000000068
Exception Note:        EXC_CORPSE_NOTIFY

VM Regions Near 0x68:
--> 
    __TEXT                 000000010bf28000-000000010bf2f000 [   28K] r-x/rwx SM=COW  /Applications/QupZilla.app/Contents/MacOS/QupZilla

Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
0   org.qt-project.QtGui          	0x000000010cb46574 QPixmap::isNull() const + 4
1   org.qt-project.QtWebKit       	0x000000010da94dcb 0x10d5d3000 + 4988363
2   org.qt-project.QtWebKit       	0x000000010d9a3b5e 0x10d5d3000 + 4000606
3   org.qt-project.QtWebKit       	0x000000010da91168 0x10d5d3000 + 4972904
4   org.qt-project.QtWebKit       	0x000000010d9a2f53 0x10d5d3000 + 3997523
5   org.qt-project.QtWebKit       	0x000000010d6161fb QWebFrameAdapter::renderRelativeCoords(QPainter*, int, QRegion const&) + 1627
6   org.qt-project.QtWebKitWidgets	0x000000010c41ec35 QWebFrame::render(QPainter*, QFlags<QWebFrame::RenderLayer>, QRegion const&) + 133
7   org.qt-project.QtWebKitWidgets	0x000000010c42b3f0 QWebView::paintEvent(QPaintEvent*) + 96
8   org.qt-project.QtWidgets      	0x000000010c4d91bb QWidget::event(QEvent*) + 2123
9   org.qt-project.QtWebKitWidgets	0x000000010c42b229 QWebView::event(QEvent*) + 441
10  org.qt-project.QtWidgets      	0x000000010c49a4bb QApplicationPrivate::notify_helper(QObject*, QEvent*) + 251
11  org.qt-project.QtWidgets      	0x000000010c49d89e QApplication::notify(QObject*, QEvent*) + 8318
12  org.qt-project.QtCore         	0x000000010d212983 QCoreApplication::notifyInternal(QObject*, QEvent*) + 115
13  org.qt-project.QtWidgets      	0x000000010c4d2970 QWidgetPrivate::drawWidget(QPaintDevice*, QRegion const&, QPoint const&, int, QPainter*, QWidgetBackingStore*) + 3152
14  org.qt-project.QtWidgets      	0x000000010c4a6c9f 0x10c469000 + 253087
15  org.qt-project.QtWidgets      	0x000000010c4a7043 0x10c469000 + 254019
16  org.qt-project.QtWidgets      	0x000000010c4c965c QWidgetPrivate::syncBackingStore() + 236
17  org.qt-project.QtWidgets      	0x000000010c4d9484 QWidget::event(QEvent*) + 2836
18  org.qt-project.QtWidgets      	0x000000010c5e3741 QMainWindow::event(QEvent*) + 929
19  org.qt-project.QtWidgets      	0x000000010c49a4bb QApplicationPrivate::notify_helper(QObject*, QEvent*) + 251
20  org.qt-project.QtWidgets      	0x000000010c49d89e QApplication::notify(QObject*, QEvent*) + 8318
21  org.qt-project.QtCore         	0x000000010d21365b QCoreApplicationPrivate::sendPostedEvents(QObject*, int, QThreadData*) + 971
22  libqcocoa.dylib               	0x000000011033a91e 0x110318000 + 141598
23  libqcocoa.dylib               	0x000000011033b1a1 0x110318000 + 143777
24  com.apple.CoreFoundation      	0x00007fff95444881 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 17
25  com.apple.CoreFoundation      	0x00007fff95423fbc __CFRunLoopDoSources0 + 556
26  com.apple.CoreFoundation      	0x00007fff954234df __CFRunLoopRun + 927
27  com.apple.CoreFoundation      	0x00007fff95422ed8 CFRunLoopRunSpecific + 296
28  com.apple.HIToolbox           	0x00007fff8e673935 RunCurrentEventLoopInMode + 235
29  com.apple.HIToolbox           	0x00007fff8e67376f ReceiveNextEventCommon + 432
30  com.apple.HIToolbox           	0x00007fff8e6735af _BlockUntilNextEventMatchingListInModeWithFilter + 71
31  com.apple.AppKit              	0x00007fff8ccffefa _DPSNextEvent + 1067
32  com.apple.AppKit              	0x00007fff8ccff32a -[NSApplication _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 454
33  com.apple.AppKit              	0x00007fff8ccf3e84 -[NSApplication run] + 682
34  libqcocoa.dylib               	0x000000011033a08d 0x110318000 + 139405
35  org.qt-project.QtCore         	0x000000010d21007d QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) + 381
36  org.qt-project.QtCore         	0x000000010d212f5a QCoreApplication::exec() + 346
37  com.qupzilla.QupZilla         	0x000000010bf2daaa main + 1658
38  com.qupzilla.QupZilla         	0x000000010bf2cff4 start + 52
```

**Comment by r-a-v-a-s** [comment]
This is odd because the App frameworks folder contains the WebEngine frameworks and not the WebKit frameworks.

**Comment by r-a-v-a-s** [comment]
After running the program once via Qt Creator and crashing it via getflow.com,
the next time I would try to run the program it would crash and complain about an old profile.
It points to the Settings class, and notes that `s_settings` is null.
Running the downloaded version from the Applications folder would fix this.

I deleted QupZilla from `~/Library/Application Support`
and then pressed build/run in Creator...
suddenly I'm using the version I built!

The website will not crash it;
however, there is a warning in Creator:

`Uncaught SecurityError: Blocked a frame with origin "https://accounts.google.com" from accessing a frame with origin "https://www.getflow.com". Protocols, domains, and ports must match.`
